components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -209,16 +209,32 @@ export class SidebarChannelMenu extends React.PureComponent<Props, State> {
 
         let leaveChannel;
         if (channel.name !== Constants.DEFAULT_CHANNEL) {
-            leaveChannel = (
-                <Menu.Group>
-                    <Menu.ItemAction
-                        id={`leave-${channel.id}`}
-                        onClick={this.handleLeaveChannel}
-                        icon={<i className='icon-close'/>}
-                        text={leaveChannelText}
-                    />
-                </Menu.Group>
-            );
+            if (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL) {
+                leaveChannel =
+                    (<Menu.Group>
+                        <Menu.ItemAction
+                            id={`leave-${channel.id}`}
+                            onClick={this.handleLeaveChannel}
+                            icon={<i className='icon-close'/>}
+                            text={leaveChannelText}
+                        />
+                    </Menu.Group>);
+            } else {
+                leaveChannel = (
+                    <Menu.Group>
+                        <span
+                            onClick={this.handleLeaveChannel}
+                            id={`leave-${channel.id}`}
+                        >
+                            <span className='MenuItem__divider'/>
+                            <li className='destructive-leave-channel'>
+                                <i className='icon-close text-white'/>
+                                {leaveChannelText}
+                            </li>
+                        </span>
```

Comment:
we can use the `Menu.ItemAction` component here as well instead. It has a prop called `isDangerous` that should add the correct semantic colors to it.